packages/loader/container-loader/src/connectionManager.ts

```diff
@@ -97,9 +101,10 @@ class NoDeltaStream
 	implements IDocumentDeltaConnection, IDisposable
 {
 	clientId = clientIdNoDeltaStream;
-	claims: ITokenClaims = {
+	// eslint-disable-next-line @typescript-eslint/consistent-type-assertions
```

Comment:
Huh, I don't remember seeing any alternative way of doing type assertions in our codebase. Does this have more details about why it complains?

Reply:
If we are bypassing the type system strictness anyways by disabling the linter rule, why can't we first typecast it to `unknown` instead?

`claims = {
		scopes: [ScopeType.DocRead],
	} as unknown as ITokenClaims;`

<img width="610" alt="Screenshot 2024-06-20 at 18 43 08" src="https://github.com/microsoft/FluidFramework/assets/111468570/ae70dd18-1211-4d25-bf66-346adc4399fa">

Reply:
It's not necessary here. We have to do that when casting something that Typescript already thinks has some type, and that type is incompatible with the one we want it to use. In this case the object has no "pre-determined" type so we can just tell Typescript the "final type" for it.

The rule as I understood it is about something else, about not mixing `variable as Type` and `<Type>variable` which apparently mean the same thing. And I don't remember seeing the second pattern anywhere in our repo, thus my question of what did eslint consider an alternate kind of type assertion.

Reply:
The rule being disabled here is one specifically intended to avoid as-casting. It prefers:

```typescript
const foo: Foo = 5;
```

over

```typescript
const foo = 5 as Foo;
```

Reply:
In the previous code, we were doing both, but the `as any` was pointless, since we were assigning to a variable typed as `ITokenClaims`. The pattern I introduced keeps the sketchy as-cast, but casts directly to the declared type.